lib/retrieve_title.rb

```diff
@@ -11,6 +11,9 @@ def self.crawl(url)
 
   def self.extract_title(html, encoding = nil)
     title = nil
+    if html =~ /<title>/ && html !~ /<\/title>/
```

Comment:
no need for 2 regexes here :) 

`/<title>.*<\/title>/`

Reply:
The above regex will create an issue when `title` tag is missing and we need to fall back on `og:title` for extracting title.